jupyterlab_bigquery/src/components/list_items_panel/service/bigquery_service.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  ServerProxyTransportService,
+  TransportService,
+  GET,
+} from 'gcp_jupyterlab_shared';
+import { Project } from './list_items';
+
+const BIGQUERY = 'https://content-bigquery.googleapis.com/bigquery/v2';
+
+interface DatasetReference {
+  datasetId: string;
+  projectId: string;
+}
+
+interface Dataset {
+  kind: string;
+  id: string;
+  datasetReference: DatasetReference;
+  labels;
+  friendlyName: string;
+  location: string;
+}
+
+interface DatasetList {
+  kind: string;
+  etag: string;
+  nextPageToken: string;
+  datasets: Dataset[];
+}
+
+/**
+ * Class to interact with BigQuery services.
+ */
+export class BigQueryService {
+  constructor(private _transportService: TransportService) {}
+
+  async listDatasets(project: Project): Promise<Project> {
+    const _transportService = new ServerProxyTransportService();
+    this._transportService;
+    return _transportService
+      .submit({
+        path: `${BIGQUERY}/projects/${project.id}/datasets`,
+        method: GET,
+        headers: { 'Content-Type': 'application/json' },
+        params: {
+          all: true,
+        },
+      })
+      .then(response => {
+        const datasetIds = [];
+        const datasets = {};
+
+        (response.result as DatasetList).datasets.forEach(dataset => {
```

Comment:
Prefer the `for (const x of ...)` style of loop iteration over `forEach`.

go/tsstyle#iterating-containers

Reply:
Done.